baselines/diabetic_retinopathy_detection/experiments/config/imagenet21k_vit_base16_finetune.py

```diff
@@ -87,20 +90,54 @@ def get_config():
   # This is "no head" fine-tuning, which we use by default
   config.model.representation_size = None
 
-  # Optimizer section
+  # Preprocessing
+
+  # Input resolution of each retina image. (Default: 512)
+  config.pp_input_res = 512  # pylint: disable=invalid-name
+  pp_common = f'|onehot({config.num_classes})'
+  config.pp_train = (
+      f'diabetic_retinopathy_preprocess({config.pp_input_res})' + pp_common)
+  config.pp_eval = (
+      f'diabetic_retinopathy_preprocess({config.pp_input_res})' + pp_common)
+
+  # Training Misc
+  config.batch_size = 512  # using TPUv3-64
+  config.seed = 0  # Random seed.
+  config.shuffle_buffer_size = 15_000  # Per host, so small-ish is ok.
+
+  # Optimization
   config.optim_name = 'Momentum'
   config.optim = ml_collections.ConfigDict()
   config.loss = 'softmax_xent'  # or 'sigmoid_xent'
-  # config.loss = 'sigmoid_xent'
-
-  # config.lr = ml_collections.ConfigDict()
+  config.lr = ml_collections.ConfigDict()
+  config.grad_clip_norm = 1.0  # Gradient clipping threshold.
+  config.weight_decay = None  # No explicit weight decay.
+  config.lr.base = 0.003
+  config.lr.decay_type = 'linear'
+
+  # The dataset is imbalanced (e.g., in Country Shift, we have 19.6%, 18.8%,
+  # 19.2% positive examples in train, val, test respectively).
+  # None (default) will not perform any loss reweighting.
+  # 'constant' will use the train proportions to reweight the binary cross
+  #   entropy loss.
+  # 'minibatch' will use the proportions of each minibatch to reweight the loss.
+  config.class_reweight_mode = None
+
+  # Evaluation Misc
+  config.only_eval = False  # Disables training, only evaluates the model
+  config.use_validation = True  # Whether to use a validation split
+  config.use_test = True  # Whether to use a test split
+
+  # Step Counts
+
+  # Varied together for wandb sweep compatibility.
+  config.total_and_warmup_steps = (10_000, 500)
```

Comment:
Does wandb support a zip-type iterator instead of just the product of all possible sweeps? It's preferable to keep these two hparams as two config settings instead of a tuple.

Reply:
Following discussion, let's add a TODO here.